packages/core/src/messagingProtocols/proteus/ProteusService/ProteusService.ts

```diff
@@ -173,4 +179,47 @@ export class ProteusService {
       state: response.errored ? PayloadBundleState.CANCELLED : PayloadBundleState.OUTGOING_SENT,
     };
   }
+
+  public async encrypt(
+    plainText: Uint8Array,
+    recipients: UserPreKeyBundleMap | UserClients,
+    domain: string = '',
+  ): Promise<{missing: UserClients; encrypted: OTRRecipients<Uint8Array>}> {
+    const {sessions, userClients} = await getSessionsAndClientsFromRecipients({
+      recipients,
+      domain,
+      apiClient: this.apiClient,
+      coreCryptoClient: this.coreCryptoClient,
+      cryptographyService: this.cryptographyService,
+      logger: this.logger,
+    });
```

Comment:
Humm I might be wrong, but I have the feeling that we are creating sessions for all the users everytime we encrypt something. This feels a little wrong here 🤔 

Creating session should happen only for the sessions that do not exist yet. (the `getSessionAndClientsFromRecipients` seems to never be checking if the session already exists)

Reply:
The check is inside `createSessionsFromPreKeys` method